test/HttpRequestTests.cpp

```diff
@@ -507,8 +507,8 @@ void HttpRequestTests::test500GetStatuses()
     };
 
     // Start the polling thread.
-    SocketPoll pollThread("AsyncReqPoll");
-    pollThread.startThread();
+    std::shared_ptr<SocketPoll> pollThread = std::make_shared<SocketPoll>("AsyncReqPoll");
```

Comment:
One more question: is the intention that we now hold SocketPoll by shared_ptr everywhere? Or did you just intend to change the places which are needed for asyncConnect()?

I see a few places that still don't use shared_ptr for SocketPoll and I wonder if that's OK or is a problem. Thanks.

Reply:
OK, I suggest to have that in a follow-up PR then.